### PR TITLE
refactor: avoid prop drilling with auth context

### PR DESCRIPTION
**Pull request by falconx**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -91,113 +78,126 @@ const cache = new InMemoryCache({
...
-        mapCenter {
-          lat
-          lng
-        }
-        userData {
-          loggedIn
-          name
-        }
+ReactDOM.render(
+  <AuthProvider>
```
Comment (marxian) [inline src/index.js:82]: I'm confused about doing the apolloClient setup inside the dom renderer.

I'm struggling to put my finger on quite why but it feels... odd somehow?

Am I worrying over nothing?

Comment (falconx) [reply src/index.js:82]: I know what you mean. It feels weird to me, so much so I messaged Steve about the render prop approach. Do you know of another solution?

I guess we could hide the setup with something like this:

`index.js`:
```
<AuthProvider>
  <MyApolloProvider>
```

`MyApolloProvider.js`:
```
const auth = useAuth();
const client = new ApolloClient({ .. });

<ApolloClient client={client}>
```

Comment (falconx) [reply src/index.js:82]: Perhaps like this https://github.com/neontribe/gbptm/compare/auth...auth-app?expand=1

Comment (marxian) [reply src/index.js:82]: Well that's certainly less disturbing!

**Pull request merged**